exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/TransactionInvoker.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.runtime;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.exonum.binding.core.service.Service;
+import com.exonum.binding.core.transaction.TransactionContext;
+import com.exonum.binding.core.transaction.TransactionExecutionException;
+import com.exonum.core.messages.Runtime.ErrorKind;
+import com.exonum.core.messages.Runtime.ExecutionError;
+import java.lang.invoke.MethodHandle;
+import java.util.Map;
+
+/**
+ * Stores ids of transaction methods and their method handles of a corresponding service.
+ */
+final class TransactionInvoker {
+  private final Service service;
+  private final Map<Integer, MethodHandle> transactionMethods;
+
+  TransactionInvoker(Service service) {
+    this.service = service;
+    this.transactionMethods =
+        TransactionMethodExtractor.extractTransactionMethods(service.getClass());
+  }
+
+  /**
+   * Execute the transaction, possibly modifying the blockchain state.
```

Comment:
I feel that the copy-pasted _user_-facing documentation is not appropriate here.
This one may be more concise and bring a different message (what it *does* for which Service
implementations, not what it *shall* do as an interface implementation). The spec is different, and the target audience is completely different.